csharp/src/Ice/OutgoingResponseFrame.cs

```diff
@@ -199,6 +216,9 @@ public static OutgoingResponseFrame WithVoidReturnValue(Current current)
                 }
                 else
                 {
+                    // In this case we always have an encapsulation it is either an ice1 response with result type
```

Comment:
An ice1 response or reply has ReplyStatus OK, which has the same value as an ice2 response with ResultType Success.
And here, the only thing that changes is the encapsulation header. "Rewrite with the new encoding" sounds like we're transcoding.